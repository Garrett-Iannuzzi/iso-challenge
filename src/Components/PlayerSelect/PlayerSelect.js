import React, { Component } from 'react';
import './PlayerSelect.scss';
import { connect } from 'react-redux';
import { selectPlayer } from '../../actions/actions';

export class PlayerSelect extends Component {
  constructor(props) {
    super(props)
    this.state ={
      selectedPlayer: ''
    }
  }

  handleChange = (e) => {
    const player = e.target.value
    this.setState({ selectedPlayer: player })
    this.props.selectPlayer(player, this.props.team, this.props.index)
  }


  render() {
    return(
      <select value={this.state.selectedPlayer} onChange={ (e) => this.handleChange(e) }>
        {this.props.players}
      </select>
    )
  }
}

export const mapDispatchToProps = dispatch => ({
  selectPlayer: (player, team, index) => dispatch(selectPlayer(player, team, index)) 
})

export default connect(null, mapDispatchToProps)(PlayerSelect)