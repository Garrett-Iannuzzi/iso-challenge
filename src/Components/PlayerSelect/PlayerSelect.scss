select {
  padding-left: 10px;
  font-size: 1em;
  margin: 10px 10px;
  margin-top: 0;
  height: 30px;
  width: 80%;
}