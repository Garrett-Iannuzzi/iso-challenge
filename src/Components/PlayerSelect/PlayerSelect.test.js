import React from 'react';
import { PlayerSelect } from './PlayerSelect';
import { shallow } from 'enzyme';

describe('PlayerSelect', () => {

  let wrapper

  describe('PlayerSelect Component', () => {

    beforeEach(() => {
      wrapper = shallow(<PlayerSelect />)
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    });
  });

})