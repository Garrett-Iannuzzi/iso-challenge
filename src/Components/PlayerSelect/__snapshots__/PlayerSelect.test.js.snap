// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerSelect PlayerSelect Component should match the snapshot 1`] = `
<select
  onChange={[Function]}
  value=""
/>
`;
